network/manager.go

```diff
@@ -395,9 +395,21 @@ func (nm *networkManager) CreateEndpoint(cli apipaClient, networkID string, epIn
 	if err != nil {
 		return err
 	}
+	// any error after this point should also clean up the endpoint we created above
+	defer func() {
+		if err != nil {
+			logger.Error("Create endpoint failure", zap.Error(err))
+			logger.Info("Cleanup resources")
+			delErr := nw.deleteEndpoint(nm.netlink, nm.plClient, nm.netio, nm.nsClient, nm.iptablesClient, ep.Id)
+			if delErr != nil {
+				logger.Error("Deleting endpoint after create endpoint failure failed with", zap.Error(delErr))
```

Comment:
Isn't this an error log for if deleteEndpoint() fails?

Reply:
yes

Reply:
This log looks like it is for a create endpoint failure while "Deleting" is making me think it succeeding or is actively doing something. Perhaps we can reference that this delete is a cleanup of a failed create endpoint?
```suggestion
				logger.Error("Create endpoint cleanup with delete endpoint failed with", zap.Error(delErr))
```

 I don't think just stating `delete endpoint failed` is good enough here so I understand the verbosity.